src/main/java/org/cascadebot/cascadebot/commands/management/permission/GroupPermissionMoveSubCommand.java

```diff
@@ -0,0 +1,113 @@
+
+    @Override
+    public void onCommand(Member sender, CommandContext context) {
+        if (context.getData().getPermissions().getMode() == GuildPermissions.PermissionMode.MOST_RESTRICTIVE) {
+            context.getTypedMessaging().replyDanger("Cannot move groups in most restrictive mode");
+            return;
+        }
+
+        if (context.getArgs().length < 1) {
+            context.getUIMessaging().replyUsage(this, "groupperms");
+            return;
+        }
+
+        PermissionCommandUtils.tryGetGroupFromString(context, context.getArg(0), group -> {
+            if (context.getArgs().length > 1 && context.isArgInteger(1)) {
+                context.getData().getPermissions().moveGroup(group, context.getArgAsInteger(1));
+                context.getTypedMessaging().replySuccess("Moved group " + group.getName() + " to position " + context.getArg(1));
+                return;
+            }
+
+            AtomicInteger currIndex = new AtomicInteger(context.getData().getPermissions().getGroups().indexOf(group));
+            ButtonGroup buttonGroup = new ButtonGroup(sender.getUser().getIdLong(), context.getChannel().getIdLong(), context.getGuild().getIdLong());
+            buttonGroup.addButton(new Button.UnicodeButton(UnicodeConstants.ARROW_UP, (runner, channel, message) -> {
+                if (buttonGroup.getOwner().getUser().getIdLong() != runner.getUser().getIdLong()) {
+                    return;
+                }
+                context.getData().getPermissions().moveGroup(context.getData().getPermissions().getGroups().get(currIndex.get()), currIndex.get() - 1);
+                currIndex.addAndGet(-1);
+                message.editMessage(getGroupsList(group, context.getData().getPermissions().getGroups())).queue();
+            }));
+            buttonGroup.addButton(new Button.UnicodeButton(UnicodeConstants.ARROW_DOWN, (runner, channel, message) -> {
+                if (buttonGroup.getOwner().getUser().getIdLong() != runner.getUser().getIdLong()) {
+                    return;
+                }
+                context.getData().getPermissions().moveGroup(context.getData().getPermissions().getGroups().get(currIndex.get()), currIndex.get() + 1);
+                currIndex.addAndGet(1);
+                message.editMessage(getGroupsList(group, context.getData().getPermissions().getGroups())).queue();
+            }));
+
+            context.getUIMessaging().sendButtonedMessage(getGroupsList(group, context.getData().getPermissions().getGroups()), buttonGroup);
+        }, sender.getUser().getIdLong());
+    }
+
+    public String getGroupsList(Group targetGroup, List<Group> groups) {
+        StringBuilder stringBuilder = new StringBuilder();
+
+        int index = groups.indexOf(targetGroup);
+        int min = index - 5;
+        int max = index + 5;
+
+        if (min < 0) {
+            min = 0;
+        }
+
+        if (max >= groups.size() - 1) {
+            max = groups.size() - 1;
+        }
```

Comment:
These two if statements can be removed in favour of min/max
`int min = Math.max(index - 5, 0)`
`int max = Math.min(index + 5, groups.size() - 1`